sdk-extensions/incubator/src/main/java/io/opentelemetry/sdk/extension/incubator/fileconfig/ViewFactory.java

```diff
@@ -32,13 +35,35 @@ public View create(StreamModel model, SpiHelper spiHelper, List<Closeable> close
     if (model.getDescription() != null) {
       builder.setDescription(model.getDescription());
     }
-    if (model.getAttributeKeys() != null) {
-      builder.setAttributeFilter(new HashSet<>(model.getAttributeKeys()));
+    IncludeExcludeModel attributeKeys = model.getAttributeKeys();
+    if (attributeKeys != null) {
+      addAttributeKeyFilter(builder, attributeKeys.getIncluded(), attributeKeys.getExcluded());
     }
     if (model.getAggregation() != null) {
       builder.setAggregation(
           AggregationFactory.getInstance().create(model.getAggregation(), spiHelper, closeables));
     }
     return builder.build();
   }
+
+  private static void addAttributeKeyFilter(
+      ViewBuilder builder, @Nullable List<String> included, @Nullable List<String> excluded) {
+    if (included == null && excluded == null) {
+      return;
+    }
+    if (included == null) {
+      Set<String> excludedKeys = new HashSet<>(excluded);
```

Comment:
Is it really necessary to copy into a `Set` implementation? Is the worry about duplicates? Can we just check `.contains()` on the original list param passed in?

Reply:
Its about efficiency. Believe it or not, the attribute filter we're adding here gets called on the hotpath for any matching instruments. Set contains is more efficient than list contains. Though, now that I think about it, while set contains is better in principle, I do wonder whether it holds true with small sets. I.e. is there a tipping point where its faster to iterate over the entries? 🤔 

Anyway, paying a tiny list -> set translation tax one time during initialization is a good move.